src/card_flows/card_flow_verify_wallet.c

```diff
@@ -57,51 +55,62 @@
+#include "card_operations.h"
+#include "constant_texts.h"
+#include "ui_instruction.h"
 
-void tap_cards_for_verification_flow_controller() {
-  switch (flow_level.level_four) {
-    case TAP_CARD_ONE_FRONTEND:
-      tap_card_data.desktop_control = false;
-      tap_card_data.tapped_card = 0;
-      flow_level.level_four = TAP_CARD_ONE_BACKEND;
-      break;
+/*****************************************************************************
+ * EXTERN VARIABLES
+ *****************************************************************************/
 
-    case TAP_CARD_ONE_BACKEND:
-      readback_share_from_card(0);
-      break;
+/*****************************************************************************
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
 
-    case TAP_CARD_TWO_FRONTEND:
-      flow_level.level_four = TAP_CARD_TWO_BACKEND;
-      break;
+/*****************************************************************************
+ * PRIVATE TYPEDEFS
+ *****************************************************************************/
 
-    case TAP_CARD_TWO_BACKEND:
-      readback_share_from_card(1);
-      break;
+/*****************************************************************************
+ * STATIC FUNCTION PROTOTYPES
+ *****************************************************************************/
 
-    case TAP_CARD_THREE_FRONTEND:
-      flow_level.level_four = TAP_CARD_THREE_BACKEND;
-      break;
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
 
-    case TAP_CARD_THREE_BACKEND:
-      readback_share_from_card(2);
-      break;
+/*****************************************************************************
+ * GLOBAL VARIABLES
+ *****************************************************************************/
 
-    case TAP_CARD_FOUR_FRONTEND:
-      flow_level.level_four = TAP_CARD_FOUR_BACKEND;
-      break;
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
 
-    case TAP_CARD_FOUR_BACKEND:
-      readback_share_from_card(3);
-      break;
+/*****************************************************************************
+ * GLOBAL FUNCTIONS
+ *****************************************************************************/
+bool card_flow_verify_wallet(void) {
+  uint8_t card_number = 0;
+  bool card_write_read_status = false;
+
+  for (card_number = 1; card_number <= 4; card_number++) {
+    char display[40];
+    snprintf(display, sizeof(display), UI_TEXT_TAP_CARD, card_number);
 
-    default:
-      LOG_CRITICAL("###31");
-      reset_flow_level();
+    instruction_scr_init(ui_text_place_card_below, display);
+
+    if (!read_card_share(card_number - 1, display, ui_text_place_card_below)) {
```

Comment:
Add a todo to handle incorrect pin and locked wallet cases.

Reply:
Added a TODO, we will pick it up when we're updating wallet verification flow